src/metatrain/utils/additive/__init__.py

```diff
@@ -1,3 +1,4 @@
-from .composition import CompositionModel  # noqa: F401
+from .composition import MetatrainCompositionModel  # noqa: F401
+from .deprecated_composition import CompositionModel  # noqa: F401
```

Comment:
Could you add a warning in `CompositionModel` that this is deprecated and people should move to `MetatrainCompositionModel`? 

And maybe have `MetatrainCompositionModel` named `CompositionModel`, the other one `OldCompositionModel` and rename on import where relevant